gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/replication/ConfigBasedMultiDatasets.java

```diff
@@ -161,35 +162,38 @@ private void generateDatasetInPushMode(ReplicationConfiguration rc, URI executio
     }// outer for loop ends
   }
 
-  private void generateDatasetInPullMode(ReplicationConfiguration rc, URI executionClusterURI){
-    if(rc.getCopyMode()== ReplicationCopyMode.PUSH){
-      log.info("Skip process push mode dataset with meta data{} as job level property specify pull mode ", rc.getMetaData());
+  private void generateDatasetInPullMode(ReplicationConfiguration replicationConfiguration, URI executionClusterURI) {
+    if (replicationConfiguration.getCopyMode() == ReplicationCopyMode.PUSH) {
+      log.info("Skip process push mode dataset with meta data{} as job level property specify pull mode ",
+          replicationConfiguration.getMetaData());
       return;
     }
 
     // PULL mode
-    CopyRouteGenerator cpGen = rc.getCopyRouteGenerator();
+    CopyRouteGenerator cpGen = replicationConfiguration.getCopyRouteGenerator();
     /**
      * Replicas comes from the 'List' which will normally be set in gobblin.replicas
      * Basically this is all possible destination for this replication job.
      */
-    List<EndPoint> replicas = rc.getReplicas();
-    for(EndPoint replica: replicas){
-      // Only pull the data from current execution cluster
-      if(needGenerateCopyEntity(replica, executionClusterURI)){
+    List<EndPoint> replicas = replicationConfiguration.getReplicas();
+    for (EndPoint replica : replicas) {
+      log.debug(
+          "[ConfigBasedMultiDatasets] Evaluating replica:" + replica.getEndPointName() + " for necessity to pull.");
```

Comment:
change log message to log.debug(".... Evaluating if replica " + replica.getEndPointName() + " needs to be pulled.")